src/shogun/machine/GeneralizedLinearMachine.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * STILL A WORK IN PROGRESS
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Tej Sukhatme
+ */
+
+// #include <rxcpp/rx-lite.hpp>
+// #include <shogun/features/DotFeatures.h>
+// #include <shogun/labels/Labels.h>
+// #include <shogun/labels/RegressionLabels.h>
+#include <shogun/machine/GeneralizedLinearMachine.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+// #include <utility>
+#include <math.h>
+#include <random>
+
+using namespace shogun;
+
+GeneralizedLinearMachine::GeneralizedLinearMachine(): Machine()
+{
+	init();
+}
+
+GeneralizedLinearMachine::GeneralizedLinearMachine(const std::shared_ptr<GeneralizedLinearMachine>& machine) : Machine()
+{
+	init();
+	require(machine, "No machine provided.");
+
+	auto w = machine->get_w();
+	auto w_clone = w.clone();
+	set_w(w_clone);
+	set_bias(machine->get_bias());
+}
+
+void GeneralizedLinearMachine::init()
+{
+	m_bias = 0;
+	features = NULL;
+	m_w = NULL;
+
+	distribution= POISSON;
+	m_alpha=0.5;
+	m_tau=NULL;
+	m_lambda=0.1;
+	m_learning_rate=2e-1;
+	m_max_iter=1000;
+	m_tolerance=1e-6;
+	m_eta=2.0;
+	m_fit_intercept=true;
+}
+
+
+GeneralizedLinearMachine::~GeneralizedLinearMachine()
+{
+
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::predict(SGMatrix<float64_t> X)
```

Comment:
yeah this is apply_....

Reply:
also takes in features, not a matrix, see the base class

Reply:
As I mentioned in the previous comment, I planned to call these methods from the overridden train_machine() and apply_regression() methods after converting the features to matrices and vectors and type checking and all the other checks to make sure the input is of the right format, length, etc which would make the code cleaner as there are supposed to be several different types of Distributions in this GLM class.
Should I continue in this way? Or should I move all the code to the train_machine() and apply_regression() method?

Reply:
ok let's see how it plays out. Though I think this is still a bit messy/confusing as it is right now